client/components/signup-form.jsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { Container, Row, Col, Form, Button } from 'react-bootstrap';
+
+export default class SignUpForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: '',
+      status: ''
+    };
+    this.handleChangeUsername = this.handleChangeUsername.bind(this);
+    this.handleChangePassword = this.handleChangePassword.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChangeUsername(e) {
+    this.setState({
+      username: e.target.value
+    });
+  }
+
+  handleChangePassword(e) {
+    this.setState({
+      password: e.target.value
+    });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    fetch('/api/auth/sign-up', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(this.state)
+    })
+      .then(res => {
+        if (res.status === 201) {
+          res.json()
+            .then(data => {
+              this.setState({
+                status: `Welcome ${data.username}! Your account has been created.`
+              });
+              e.target.reset();
+            })
+            .catch(err => console.error(err));
+        } else {
+          res.json()
+            .then(data => {
+              this.setState({
+                status: data.error
+              });
+            })
+            .catch(err => console.error(err));
+        }
```

Comment:
since you are using the `error` property of the response, you can ignore the status code.
```js
.then(res => res.json())
.then(data => {
  if (data.error) {
  } else {
  }
})